core/egov-searcher/src/main/java/org/egov/search/service/SearchService.java

```diff
@@ -103,4 +103,28 @@ private String formatResult(List<String> maps, Definition searchDefinition, Sear
 		return documentContext.jsonString().toString();
 		
 	}
+
+	/**+
+	 *
+	 * @param searchCriteriafromBody :- search criteria coming from request body
+	 * @param searchCriteriafromUrl	:- search criteria coming from url query params
+	 * @return	search criteria with combination of both criterias
+	 */
+	public HashMap<String,String> fetchSearchCriteria(HashMap<String,String> searchCriteriafromBody,HashMap<String,String>searchCriteriafromUrl)
+	{
+		if((searchCriteriafromBody==null)&&((searchCriteriafromUrl==null)||(searchCriteriafromUrl.isEmpty())))
```

Comment:
Insufficient check. Modify this -
if(((searchCriteriafromBody != null) && (!searchCriteriafromBody.isEmpty())) && ((searchCriteriafromUrl != null) && (!searchCriteriafromUrl.isEmpty()))){

//INSERT ALL YOUR LOGIC.

} else{
return null;
}